tools/bag_processing/scripts/convert_bayer_bag_to_color.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# Copyright (c) 2017, United States Government, as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+#
+# All rights reserved.
+#
+# The Astrobee platform is licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+"""
+Converts bayer encoded images from the provided bagfile to grayscale images in a new bagfile.
+"""
+
+import argparse
+import os
+import sys
+
+import cv2
+import rosbag
+import rospy
+import utilities
+from cv_bridge import CvBridge, CvBridgeError
+from sensor_msgs.msg import Image
+
+
+def convert_bayer_to_color(
+    bagfile, bayer_image_topic, color_image_topic, save_all_topics=False
+):
+    bridge = CvBridge()
+    topics = None if save_all_topics else [bayer_image_topic]
+    output_bag_name = os.path.splitext(bagfile)[0] + "_color.bag"
+    output_bag = rosbag.Bag(output_bag_name, "w")
+
+    with rosbag.Bag(bagfile, "r") as bag:
+        for topic, msg, t in bag.read_messages(topics):
+            if topic == bayer_image_topic:
+                try:
+                    image = bridge.imgmsg_to_cv2(msg, "bayer_grbg8")
+                except (CvBridgeError) as e:
+                    print(e)
+                color_image = cv2.cvtColor(image, cv2.COLOR_BAYER_GB2BGR)
+                color_image_msg = bridge.cv2_to_imgmsg(color_image, encoding="bgr8")
+                color_image_msg.header = msg.header
+                output_bag.write(color_image_topic, color_image_msg, t)
+            # Also save the bayer topic unaltered
```

Comment:
Hi @rsoussan ! In the bag converter for the greyscale, there is an `else` here which means that the original bayer message isn't saved. Just wondering if it is by design given that if the greyscale script is executed first, color image extraction is not possible.

Reply:
Oh interesting, yeah I didn't think about the use case at that time.  Maybe we should add an option to save the original image as well, so users can decide if they want smaller bags or larger bags with both images

Reply:
We could also merge the two scripts here, and pass an option to do grayscale or color conversion

Reply:
Just updated it with the changes :)

Reply:
Looks good! Just one small comment regarding the arg parse options